shadows/framework/src/main/java/org/robolectric/shadows/ShadowFileProvider.java

```diff
@@ -0,0 +1,35 @@
+package org.robolectric.shadows;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Build;
+import androidx.annotation.NonNull;
+import java.io.File;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+/**
+ * A shadow implementation of androidx.core.content.FileProvider.
+ *
+ * <p>Currently only supports getUriForFile(). The returned Uri will always have "content" as its
+ * scheme, authority matching to that of the given argument, and path matching to the absolute path
+ * of the given file instance.
+ */
+@Implements(
+    className = "androidx.core.content.FileProvider",
```

Comment:
The shadow framework only shadows Android framework classes not including androidx. Could you shadow Android framework part that AndroidX FileProvider uses?

Reply:
Thanks for reviewing! Your comment made me realize the true root cause of #2199–well, at least as I experienced it anyways. It seems like the existing shadows are already adequate to support `FileProvider` as it is.

https://github.com/robolectric/robolectric/issues/2199#issuecomment-218113245 in my case was caused by a misconfigured Manifest, not lack of shadowed behavior. I'll be closing this PR soon. :)